### DISCUSSION TITLE
Alt+Tab needs better responsiveness for mouse inputs

### DISCUSSION THREAD

**Discussion by Rya3D** [linuxmint/discussions#484] [Ideas]
Alt+Tab on all the modes except the 3D options have a lag/delay for when the mouse is able to interact with the Icons/Thumbnails. 

The 3D options remove the delay but I very much dislike the look of them an defeat the purpose of why I'm using my mouse, since the windows become obstructed by other windows or just become invisible until tabbed into using my mouse to click on them becomes difficult or impossible to do. Is there a way for me to fix/change this on my end?

**Comment by GalacticSun** [comment]
I've noticed this as well, it's a bit sluggish for the cursor to interact with the alt-tab window.  And there is just the slightest delay when trying to bring up alt-tab in general, always has been on LM it seems.  Maybe I was spoiled with this back when I used Mac OS, their alt-tab was (is?) so responsive and fast.

**Reply by eioo** [reply]
I configured this in the "Windows" settings, it had something silly like 100ms as the default value (why?).

1. Open start menu
2. Type "windows", 
3. Select "Alt-tab" tab
4. Enter "0" to "Delay before displaying the alt-tab switcher (milliseconds)"

![image](https://github.com/user-attachments/assets/36bd4eb2-faa1-4f9d-a341-4d00b17e8b66)

**Reply by GalacticSun** [reply]
> I configured this in the "Windows" settings, it had something silly like 100ms as the default value (why?).
> 
>     1. Open start menu
> 
>     2. Type "windows",
> 
>     3. Select "Alt-tab" tab
> 
>     4. Enter "0" to "Delay before displaying the alt-tab switcher (milliseconds)"

Thanks.  Yeah, for me, I'd already done that.

**Comment by eioo** [comment]
For me the alt+tab menu opens immediately.

Problem is that windows do not get selected until some window has had mouse over it, in other words have to hover over window twice for it to get selected. I've demonstrated it in this video:

https://github.com/user-attachments/assets/e82a78da-0aba-48e4-b421-a567131e23a3

Hope there was solution for this, very annoying.

**Reply by 100savage** [reply]
If you select alt+tab again it will cycle through the open windows

**Reply by eioo** [reply]
This I am aware of, the mouse is the one that's not efficient to use.

**Comment by eioo** [comment]
Managed to fix my issue related to the mouse being unresponsive at the beginning.

Have to update this file: `/usr/share/cinnamon/js/ui/appSwitcher/appSwitcher.js` and restart Cinnamon with `Alt + F2 -> "r" -> Enter`.

```diff
diff --git a/appSwitcher.js b/appSwitcher.js
index 16c2378..7b77353 100644
--- a/appSwitcher.js
+++ b/appSwitcher.js
@@ -364,7 +364,7 @@ AppSwitcher.prototype = {
     },
 
     _disableHover: function () {
-        this._mouseActive = false;
+        this._mouseActive = true;
 
         if (this._motionTimeoutId != 0)
             Mainloop.source_remove(this._motionTimeoutId);
```

Reason why mouse hovering is initially disabled is mentioned in the same file:

```js
// Initially disable hover so we ignore the enter-event if
// the switcher appears underneath the current pointer location
```

Personally I don't use enter and don't mind mouse hovering being active as soon as the switcher appears.

Found this thread to be helpful: https://github.com/linuxmint/cinnamon/issues/8994#issuecomment-781540415

**Reply by GalacticSun** [reply]
Thank you, @eioo!  That fixed this issue for me.  And, you articulated the problem far better than I could and that video explains it very well.  This alt tab issue with the mouse has bothered me ever since I switched to LM a few years ago.  You're a genius!  Much appreciated.

The issue with alt-tab not coming up right away had to do with, I believe, having far too many browser windows open for my machine.  So I closed most of them.  But this mouse issue was just as bothersome to me, so thank you again for the fix.

**Comment by GalacticSun** [comment]
By the way, I think I understand at least part of the reason as to why they may have initially made it not accept the mouse input right away: Because it can make you select another window too quickly before you ever know you have selected it if your mouse is where the Alt-Tabs trip comes up.  This is fine with me, though, since it's better than the way it has been in Cinnamon before @eioo's fix.

But, I believe, a potential enhancement might be to retain the "initial window selection" in a variable in the code somewhere, so when the mouse moves outside of the Alt-Tab strip, the selected window goes back to being that initial one on the left-hand side (the window, next to your currently focused/active one, that was initially selected when you brought up Alt-Tab in the first place).  Then, if you move your mouse over the Alt-Tab strip again, it selects the window under the mouse, but then if you move your mouse outside of the Alt-Tab strip again, the selected window goes back to being the default one on the left-hand-side again.  Hopefully my description makes sense.  Not sure how to fix this in the code, but it's just an idea.
